### PR TITLE
chore(site): improve site

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- [x] 合并 #39043 新建的 SiteContext
- [x] 将主题(theme) 添加到 SiteContext 中
  - 和主题相关的可以不用通过 url 获取，统一使用 context 获取
- [x] 在 SiteContext 提供修改方法 updateSiteCinfig 功能方法
    - 提供给设置 direction， theme 时使用
- [x] 移除右下角默认主题切换按钮 (~与 #39534 重复了~, 已解决冲突
- [x] 修复 组件总览点击进入组件详情时，url 参数丢失问题
- [x] 修复 组件 Demo 展开代码图标根据主题切换逻辑问题
- [x] 修复暗色主题下，组件汇总页搜索输入框颜色错误问题
- [x] 修复暗色主题下，顶部语言切换字体颜色错误问题

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    -       |
| 🇨🇳 Chinese |    -       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: .dumi/theme/builtins/Previewer/fromDumiProps.tsx [hunk 1]
```diff
@@ -89,7 +88,6 @@ export default function fromDumiProps<P extends object>(
       expand,
       // FIXME: confirm is there has any case?
       highlightedStyle: '',
-      theme: searchParams.get('theme'),
```
Comment (Wxh16144) [inline .dumi/theme/builtins/Previewer/fromDumiProps.tsx:92]: https://user-images.githubusercontent.com/32004925/207113453-9d19360b-84e2-4848-b5e8-05daa793b16f.mp4

**Comment by Wxh16144** [comment]
https://preview-39504-ant-design.surge.sh/components/button?theme=dark&theme=compact#components-button-demo-ghost 这个地址会被吞 `?theme=drak`

**Comment by MadCcc** [comment]
PR 里写下改动点？

File: .dumi/theme/builtins/Previewer/fromDumiProps.tsx (hunk 1 above)
Comment (afc163) [reply .dumi/theme/builtins/Previewer/fromDumiProps.tsx:92]: 问题是啥？

Comment (Wxh16144) [reply .dumi/theme/builtins/Previewer/fromDumiProps.tsx:92]: 因为 url 里面的 theme 会有多个， 比如 url=xxx?theme=compact&theme=dark 通过 `theme: searchParams.get('theme')` 只会获取第一个

**Comment by Wxh16144** [comment]
基本上就这些改动吧，后面再有调整，我重新开 PR。 改动太多了 Review 有点辛苦

**Review by MadCcc — APPROVED** [review]

**Comment by afc163** [comment]
@Wxh16144 这个引起了 https://github.com/ant-design/ant-design/pull/39687#issuecomment-1360828454 的问题。

**Pull request merged**